mlflow/entities/trace_info.py

```diff
@@ -83,7 +83,7 @@ def to_dict(self):
         Update status field to the string value for serialization.
         """
         trace_info_dict = asdict(self)
-        trace_info_dict["status"] = self.status.value
```

Comment:
Following up from our thread offline, the type of `self.status` in `TraceInfo` is `TraceStatus`, so the preexisting logic should in theory be correct. My guess is that `TraceInfo` is being constructed with a string value of `TraceStatus` when a trace is read from the backend. 

The problem seems to lie in https://github.com/mlflow/mlflow/blob/f54297ded22bb3973d59aa9c39c8abed90d83016/mlflow/store/tracking/rest_store.py#L338 -> https://github.com/mlflow/mlflow/blob/f54297ded22bb3973d59aa9c39c8abed90d83016/mlflow/entities/trace_info.py#L75 -> https://github.com/mlflow/mlflow/blob/f54297ded22bb3973d59aa9c39c8abed90d83016/mlflow/entities/trace_status.py#L24

here, we see that `TraceStatus.from_proto()` doesn't return a `TraceStatus` like it should. Instead, it returns a string. We need to fix the implementation of `TraceStatus.from_proto()` and then add test coverage verifying that a `TraceStatus` constructed from a proto can be converted to dict / JSON and back.

Reply:
Yeah! I realized that the fix was superficial and hence incorrect. Fixed the from_proto hydration and added a bunch of tests